internal/crypto/pbkdf.go

```diff
@@ -0,0 +1,23 @@
+package crypto
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"golang.org/x/crypto/pbkdf2"
+)
+
+// The NIST recommended iterations for PBKDF2 with SHA256 hash is 600,000
+const pbkdf2Sha256Iterations = 1<<20 - 1<<18 // 786,432
+
+// The NIST recommended minimum size for a salt to be used for pbkdf2
+const minPbkdfSaltSize = 16 // size in bytes == 128 bits
```

Comment:
`A good rule of thumb is to use a salt that is the same size as the output of the hash function. For example, the output of SHA256 is 256 bits (32 bytes), so the salt should be at least 32 random bytes.`

REF: https://crackstation.net/hashing-security.htm

Should not this be 32 instead for both scrypt and pbkdf2 ?

Reply:
I believe throughout the code we pass in a 32 byte salt so I can change this to make that the minimum. I picked 16 because it was the minimum recommended for NIST. I will update the comments with this information.

Reply:
Also, changing the salt size will break existing users, so it is required to be consistent with the existing code unless we have a key migration path forward.